src/model/page-config-resolver.js

```diff
@@ -292,6 +335,120 @@ class MicrodataParser {
     this.doc_ = doc;
   }
 
+  /**
+   * Returns true if access is restricted, otherwise false
+   * @param {Node} root A node that is an item of type 'NewsArticle'
+   * @return {?boolean} locked access
+   * @private
+   */
+  discoverLocked_(root) {
+    const ALREADY_SEEN = 'alreadySeenForAccessInfo';
+    const nodeList = root
+        .querySelectorAll("[itemprop='isAccessibleForFree']");
+    for (let i = 0; nodeList[i]; i++) {
+      const element = nodeList[i];
+      const content = element.content || element.textContent;
+      if (!content) {
+        continue;
+      }
+      if (content.toLowerCase() == 'true') {
+        value = true;
+      } else if (content.toLowerCase() == 'false') {
+        value = false;
+      }
+      if (this.isValidElement_(element, root, ALREADY_SEEN)) {
+        return !value;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Verifies if a node is valid based on the following
+   * - child of an item of type 'NewsArticle'
+   * - not a child of an item of type 'Section'
+   * - not seen before, marked using the alreadySeen tag
+   * @param current the node to be verified
+   * @param root the parent to track up to
+   * @param alreadySeen used to tag already visited nodes
+   * @return {!boolean} valid node
+   * @private
+   */
+  isValidElement_(current, root, alreadySeen) {
+    while (current && !current[alreadySeen]) {
+      current[alreadySeen] = true;
+      if (current.hasAttribute('itemscope')) {
+        const type = current.getAttribute('itemtype');
+        if (type == 'http://schema.org/NewsArticle') {
+          // This parent may need to be check for other nodes
+          current[alreadySeen] = false;
+          return true;
+        } else if (type.indexOf('Section') >= 0) {
+          return false;
+        }
+      }
+      current = current.parentNode;
+    }
+    return false;
+  }
+
+  /**
+   * Obtains the product ID that meets the requirements
+   * - child of an item of type 'NewsArticle'
+   * - Not a child of an item of type 'Section'
+   * - child of an item of type 'productID'
+   * @param {Node} root A node that is an item of type 'NewsArticle'
+   * @return {?string} product ID, if found
+   * @private
+   */
+  discoverProductId_(root) {
+    const ALREADY_SEEN = 'alreadySeenForProductInfo';
+    const nodeList = root
+        .querySelectorAll("[itemprop='productID']");
+    for (let i = 0; nodeList[i]; i++) {
+      const element = nodeList[i];
+      const content = element.content || element.textContent;
```

Comment:
To confirm: this could only be a `<meta>` element? B/c other elements would not have a `.content` property and you'd have to read the attribute directly instead.

Reply:
as per schema.org itemprop can be a part of span, anchor tags and meta tags.

Reply:
In this case, you'll have to drop element.content and replace it with element.getAttribute('content'). Span/anchor elements do not expose content property - so this will always be undefined.